### PR TITLE
Adding drush devel generate options.

### PR DESCRIPTION
**Pull request by dzink**

### CODE REVIEW

**Comment by lliss** [comment]
Can you make an associated change in the docs.

**Comment by lliss** [comment]
Can you provide some test instructions or sample probo file.

**Comment by tizzo** [comment]
I think this is a bridge too far. The Drupal plugin shouldn't try to do everything we can think of - just to make a main-stream Drupal deployment make sense. If people want to generate content they can just do a shell step of `drush devel-generate` - I think this goes beyond deploying and goes beyond the majority case.

@lliss thoughts?

File: lib/plugins/TaskRunner/Drupal.js [hunk 1]
```diff
@@ -212,6 +224,26 @@ class Drupal extends LAMPApp {
...
+    for (var content_type in types) {
+      if (types.hasOwnProperty(content_type)) {
+        this.script.push(`cd $SRC_DIR ; drush generate-content ${types[content_type]} --types=${content_type}`);
+      }
+    }
+  }
+
+  addScriptDevelGenerateUsers(roles) {
+    for (var role_name in roles) {
+      if (roles.hasOwnProperty(role_name)) {
```
Comment (lliss) [inline lib/plugins/TaskRunner/Drupal.js]: I always feel like something if amiss if you have to check if a key exists this way. Better would be to use _.keys (https://lodash.com/docs#keys) to create an array of keys and then use those to create the loop. I also don't feel great about the particular structure but it works. I think you'll also want to make sure the value is a number, currently you have no check for that.

**Comment by lliss** [comment]
I'm on the fence. It seems like a common enough thing to want to do, and what's the harm in adding features to the Drupal plugin? The whole purpose of the Drupal plugin is to make common tasks easier so if this makes life easier for a bunch of people, then why not? On the other hand, if we make the command support everything you can and often do with drush, then we're basically just writing a yaml interface to drush and the work never ends. If that were the case, maybe we should consider making that its own thing.

In any event, this PR is for a public open source project and so it needs a summery and some testing instructions. I'm not going through the motions of building stuff out without them.

**Comment by tizzo** [comment]
@tortillaj you're changing tags without a description of why you think @lliss was wrong to change the tag?

We should keep in mind that drush *doesn't come with devel-generate*. So this will flat out fail on fresh drupal install unless you we auto-install devel. I hadn't even thought of that - yeah closing.

**Comment by lliss** [comment]
> So this will flat out fail on fresh drupal install unless we auto-install devel. I hadn't even thought of that - yeah closing.

@tizzo True but so will features revert and we have a setting for that.

**Comment by tizzo** [comment]
You raise an excellent point… I still think this seems like a minority case and a bridge too far… but fine, reopening.

That said, if we do merge this in - what params should we support? Database updates and features reverts are easy - these are the supported options (on the one version of devel generate I checked:

````
Create content.

Arguments:
 number_nodes                              Number of nodes to generate.
 maximum_comments                          Maximum number of comments to generate.

Options:
 --feedback                                An integer representing interval for insertion rate logging. Defaults to 1000
 --kill                                    Delete all content before generating new content.
 --languages                               A comma-separated list of language codes
 --skip-fields                             A comma delimited list of fields to omit when generating random values
 --types                                   A comma delimited list of content types to create. Defaults to page,article.
````

**Comment by lliss** [comment]
@dzink Putting this back on your radar for further thoughts. I'm inclined to close this PR at this point.

**Pull request closed without merging**